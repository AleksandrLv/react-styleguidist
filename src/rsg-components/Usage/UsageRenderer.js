import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

const styles = ({ space, fontFamily, fontSize, color }) => ({
	section: {
		marginBottom: space[4],
	},
	heading: {
		marginBottom: space[1],
		color: color.base,
		fontFamily: fontFamily.base,
		fontWeight: 'normal',
		fontSize: fontSize.h4,
	},
});

export function UsageRenderer({ classes, props, methods }) {
	if (!props && !methods) {
		return null;
	}

	return (
		<div>
			{props &&
				<div className={classes.section}>
					<h4 className={classes.heading}>Props</h4>
					{props}
				</div>}
			{methods &&
				<div className={classes.section}>
					<h4 className={classes.heading}>Methods</h4>
					{methods}
				</div>}
		</div>
	);
}

UsageRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	props: PropTypes.node,
	methods: PropTypes.node,
};

export default Styled(styles)(UsageRenderer);
