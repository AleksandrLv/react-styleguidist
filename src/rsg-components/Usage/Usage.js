import React from 'react';
import PropTypes from 'prop-types';
import Props from 'rsg-components/Props';
import Methods from 'rsg-components/Methods';
import UsageRenderer from 'rsg-components/Usage/UsageRenderer';

export default function Usage(component) {
	const { props, methods } = component.props;
	return (
		<UsageRenderer
			props={props && <Props props={props} />}
			methods={methods && methods.length > 0 && <Methods methods={methods} />}
		/>
	);
}

Usage.propTypes = {
	props: PropTypes.shape({
		props: PropTypes.object,
		methods: PropTypes.array,
	}).isRequired,
};
