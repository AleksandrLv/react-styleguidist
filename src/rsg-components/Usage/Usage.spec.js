import React from 'react';
import Usage from './Usage';
import { UsageRenderer } from './UsageRenderer';

const props = {
	children: {
		type: { name: 'string' },
		required: true,
		description: 'Button label.',
	},
};
const methods = [
	{
		name: 'set',
		params: [
			{
				name: 'newValue',
				description: 'New value for the counter.',
				type: { name: 'Number' },
			},
		],
		returns: null,
		description: 'Sets the counter to a particular value.',
	},
];

describe('Usage', () => {
	it('should pass props and methods to renderer', () => {
		const actual = shallow(<Usage props={{ props, methods }} />);

		expect(actual).toMatchSnapshot();
	});
});

describe('UsageRenderer', () => {
	it('should render component renderer for component with props', () => {
		const actual = shallow(<UsageRenderer classes={{}} props={<div>props</div>} />);

		expect(actual).toMatchSnapshot();
	});

	it('should render component renderer for component with methods', () => {
		const actual = shallow(<UsageRenderer classes={{}} methods={<div>methods</div>} />);

		expect(actual).toMatchSnapshot();
	});

	it('should render nothing without props and methods', () => {
		const actual = shallow(<UsageRenderer classes={{}} />);

		expect(actual.node).toBe(null);
	});
});
