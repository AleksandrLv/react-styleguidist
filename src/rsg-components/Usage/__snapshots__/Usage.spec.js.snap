// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Usage should pass props and methods to renderer 1`] = `
<Styled(Usage)
  methods={
    <MethodsRenderer
      methods={
        Array [
          Object {
            "description": "Sets the counter to a particular value.",
            "name": "set",
            "params": Array [
              Object {
                "description": "New value for the counter.",
                "name": "newValue",
                "type": Object {
                  "name": "Number",
                },
              },
            ],
            "returns": null,
          },
        ]
      }
    />
  }
  props={
    <PropsRenderer
      props={
        Object {
          "children": Object {
            "description": "Button label.",
            "required": true,
            "type": Object {
              "name": "string",
            },
          },
        }
      }
    />
  }
/>
`;

exports[`UsageRenderer should render component renderer for component with methods 1`] = `
<div>
  <div>
    <h4>
      Methods
    </h4>
    <div>
      methods
    </div>
  </div>
</div>
`;

exports[`UsageRenderer should render component renderer for component with props 1`] = `
<div>
  <div>
    <h4>
      Props
    </h4>
    <div>
      props
    </div>
  </div>
</div>
`;
